src/fal/el/fivetran.py

```diff
@@ -0,0 +1,175 @@
+"""Synchronize with Fivetran."""
+import time
+import requests
+import datetime
+import json
+from requests.auth import HTTPBasicAuth
+from requests.exceptions import RequestException
+from dataclasses import dataclass, field
+from typing import Any, Dict, List
+from urllib.parse import urljoin
+from dateutil import parser
+from dbt.logger import GLOBAL_LOGGER as logger
+
+BASE_URL = urljoin("https://api.fivetran.com/", "v1/connectors/")
+SCHEDULE_TYPES = ["auto", "manual"]
+
+
+@dataclass
+class FivetranClient:
+    """Fivetran API client."""
+
+    api_key: str
+    api_secret: str
+    disable_schedule_on_trigger: bool = True
+    max_retries: int = 3
+    retry_delay: float = 0.25
+    _auth: Any = field(init=False)
+
+    def __post_init__(self):
+        """Set variables."""
+        self._auth = HTTPBasicAuth(self.api_key, self.api_secret)
+
+    def request(self, endpoint: str, method: str, data: str = None):
```

Comment:
it seems `request` is just used internally. Do we want to make it private? `_request`?